accelerator/proxy_apis.c

```diff
@@ -0,0 +1,352 @@
+/*
+ * Copyright (C) 2018-2019 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#include "proxy_apis.h"
+#include "utils/handles/lock.h"
+
+#define PROXY_APIS_LOGGER "proxy_apis"
+
+static logger_id_t logger_id;
+static lock_handle_t cjson_lock;
+
+void proxy_apis_logger_init() { logger_id = logger_helper_enable(PROXY_APIS_LOGGER, LOGGER_DEBUG, true); }
+
+int proxy_apis_logger_release() {
+  logger_helper_release(logger_id);
+  if (logger_helper_destroy() != RC_OK) {
+    ta_log_critical("Destroying logger failed %s.\n", PROXY_APIS_LOGGER);
+    return EXIT_FAILURE;
+  }
+
+  return 0;
+}
+
+status_t proxy_apis_lock_init() {
+  if (lock_handle_init(&cjson_lock)) {
+    return SC_CONF_LOCK_INIT;
+  }
+  return SC_OK;
+}
+
+status_t proxy_apis_lock_destroy() {
+  if (lock_handle_destroy(&cjson_lock)) {
+    return SC_CONF_LOCK_DESTROY;
+  }
+  return SC_OK;
+}
+
+status_t char_buffer_to_str(char_buffer_t* res_buff, char** json_result) {
+  status_t ret = SC_OK;
+  *json_result = (char*)malloc((res_buff->length + 1) * sizeof(char));
+  if (*json_result == NULL) {
+    ret = SC_TA_OOM;
+    ta_log_error("%s\n", "SC_TA_OOM");
+    goto done;
+  }
+  snprintf(*json_result, (res_buff->length + 1), "%s", res_buff->data);
+
+done:
+  return ret;
+}
```

Comment:
Actually we can return `status_t` directly, since nothing needs to be freed in this function:
```suggestion
status_t char_buffer_to_str(char_buffer_t* res_buff, char** json_result) {
  *json_result = (char*)malloc((res_buff->length + 1) * sizeof(char));
  if (*json_result == NULL) {
    ta_log_error("%s\n", "SC_TA_OOM");
    return SC_TA_OOM;
  }
  snprintf(*json_result, (res_buff->length + 1), "%s", res_buff->data);
  
  return SC_OK;
}
```